### PR TITLE
Problem: Msg Priority field lacks validation

### PR DESCRIPTION
**Pull request by dkliban**
Solution: validate that Msg Priority field is one of the following values:
"Emerg", "Alert", "Crit", "Err", "Warning", "Notice", "Info", "Debug"

### CODE REVIEW

File: resources/msg.go [hunk 1]
```diff
@@ -69,6 +69,16 @@ func (obj *MsgRes) Validate() error {
 			return fmt.Errorf("fields cannot begin with _")
 		}
 	}
+        ValidPriorities := []string{"Emerg", "Alert", "Crit", "Err", "Warning", "Notice", "Info", "Debug"}
```
Comment (purpleidea) [inline resources/msg.go]: uppercase things are usually public. This should start with a lowercase name.

File: resources/msg.go [hunk 2]
```diff
@@ -69,6 +69,16 @@ func (obj *MsgRes) Validate() error {
...
+        priorityIsValid := false
+        for _, element := range ValidPriorities {
+            if strings.Compare(obj.Priority, element) == 0 {
+                priorityIsValid = true
+            }
+        }
+        if priorityIsValid == false {
```
Comment (purpleidea) [inline resources/msg.go]: I think this whole approach could be simplified... How about just a switch statement: https://tour.golang.org/flowcontrol/9

File: resources/msg.go [hunk 3]
```diff
@@ -69,6 +69,16 @@ func (obj *MsgRes) Validate() error {
 			return fmt.Errorf("fields cannot begin with _")
 		}
 	}
+        ValidPriorities := []string{"Emerg", "Alert", "Crit", "Err", "Warning", "Notice", "Info", "Debug"}
+        priorityIsValid := false
+        for _, element := range ValidPriorities {
+            if strings.Compare(obj.Priority, element) == 0 {
```
Comment (purpleidea) [inline resources/msg.go]: Please read the recommendation on this methods docs: https://golang.org/pkg/strings/#Compare

File: resources/msg.go [hunk 4]
```diff
@@ -114,7 +123,6 @@ func (obj *MsgRes) journalPriority() journal.Priority {
 	case "Debug":
 		return journal.PriDebug
 	}
-	return journal.PriNotice
```
Comment (purpleidea) [inline resources/msg.go]: Why did you remove this?

File: resources/msg_test.go [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+}
+
+func TestMsgValidate2(t *testing.T) {
+        r1 := &MsgRes{
+                BaseRes: BaseRes{
+                        Name: "msg1",
+                        Kind: "msg",
+                        MetaParams: DefaultMetaParams,
+                },
+                Priority:   "Priority",
```
Comment (purpleidea) [inline resources/msg_test.go]: Tests look good! But maybe make the bad value slightly more obvious?

File: resources/msg.go (hunk 4 above)
Comment (dkliban) [inline resources/msg.go]: i made the assumption that it was there because you didn't have validation and wanted it to default to Notice when a match was not found.

File: resources/msg.go (hunk 1 above)
Comment (dkliban) [inline resources/msg.go]: ah yes!

File: resources/msg.go (hunk 3 above)
Comment (dkliban) [inline resources/msg.go]: thanks

File: resources/msg.go (hunk 2 above)
Comment (dkliban) [inline resources/msg.go]: much simpler :)

File: resources/msg.go [hunk 6]
```diff
@@ -69,6 +69,18 @@ func (obj *MsgRes) Validate() error {
 			return fmt.Errorf("fields cannot begin with _")
 		}
 	}
+	switch priority := obj.Priority; priority {
```
Comment (purpleidea) [inline resources/msg.go]: why all this instead of just `switch obj.Priority {` ?

File: resources/msg.go [hunk 7]
```diff
@@ -69,6 +69,18 @@ func (obj *MsgRes) Validate() error {
...
+	case "Emerg":
+	case "Alert":
+	case "Crit":
+	case "Err":
+	case "Warning":
+	case "Notice":
+	case "Info":
+	case "Debug":
+	default:
+		return fmt.Errorf("invalid Priority")
```
Comment (purpleidea) [inline resources/msg.go]: LGTM, maybe add a invalid Priority
```
"`%s`", obj.Priority)

to the end?

**Comment by purpleidea** [comment]
@dkliban Hey! Merged :) Sorry I didn't do this sooner, I didn't realize you had pushed a new patch! GH is ridiculously lame here and doesn't send notifications.

Thanks! LMK if you're ready for something more interesting!
